src/App/Pages/Settings/SecuritySettingsPageViewModel.cs

```diff
@@ -234,10 +256,11 @@ public async Task InitAsync()
             _supportsBiometric = await _platformUtilsService.SupportsBiometricAsync();
             _canUnlockWithBiometrics = await _vaultTimeoutService.IsBiometricLockSetAsync();
             _canUnlockWithPin = await _vaultTimeoutService.GetPinLockTypeAsync() != Core.Services.PinLockType.Disabled;
+            IsVaultTimeoutActionLockAllowed = _hasMasterPassword || _canUnlockWithBiometrics || _canUnlockWithPin;
 
             await LoadPoliciesAsync();
             await InitVaultTimeoutPickerAsync();
-            await InitVaultTimeoutActionPickerAsync();
+            await MainThread.InvokeOnMainThreadAsync(InitVaultTimeoutActionPickerAsync);
```

Comment:
🤔 The call to the main thread should be done only on the part that is needed inside the `IniVaultTimeoutActionPickerAsync()`. Where is that necessary?

Reply:
Actually i didn't needed, it was something else that had me doing it but after all the changes it works without it.